shogun-lib/src/main/java/de/terrestris/shogun/lib/controller/security/permission/BasePermissionController.java

```diff
@@ -0,0 +1,837 @@
+import de.terrestris.shogun.lib.service.BaseService;
+import de.terrestris.shogun.lib.service.GroupService;
+import de.terrestris.shogun.lib.service.UserService;
+import de.terrestris.shogun.lib.service.security.permission.GroupClassPermissionService;
+import de.terrestris.shogun.lib.service.security.permission.GroupInstancePermissionService;
+import de.terrestris.shogun.lib.service.security.permission.UserClassPermissionService;
+import de.terrestris.shogun.lib.service.security.permission.UserInstancePermissionService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.core.GenericTypeResolver;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+// TODO Specify and type extension of BaseService
+public abstract class BasePermissionController<T extends BaseService<?, S>, S extends BaseEntity> {
+
+    protected final Logger LOG = LogManager.getLogger(getClass());
+
+    @Autowired
+    protected T service;
+
+    @Autowired
+    protected MessageSource messageSource;
+
+    @Autowired
+    protected UserService userService;
+
+    @Autowired
+    protected GroupService groupService;
+
+    @Autowired
+    protected UserInstancePermissionService userInstancePermissionService;
+
+    @Autowired
+    protected UserClassPermissionService userClassPermissionService;
+
+    @Autowired
+    protected GroupInstancePermissionService groupInstancePermissionService;
+
+    @Autowired
+    protected GroupClassPermissionService groupClassPermissionService;
+
+    @GetMapping("/{id}/permissions/instance/user")
+    @ResponseStatus(HttpStatus.OK)
+    public List<UserInstancePermission> getUserInstancePermissions(@PathVariable("id") Long entityId) {
+        LOG.trace("Requested to get all user instance permissions for entity of type {} with ID {}",
+            getGenericClassName(), entityId);
+
+        try {
+            Optional<S> entity = service.findOne(entityId);
+
+            if (entity.isPresent()) {
+                List<UserInstancePermission> permissions = userInstancePermissionService.findFor(entity.get());
+
+                LOG.trace("Successfully got all user instance permissions for entity of type {} with " +
+                    "ID {} (count: {})", getGenericClassName(), entityId, permissions.size());
+
+                return permissions;
+            } else {
+                throw getEntityNotFoundException(entityId);
+            }
+        } catch (AccessDeniedException ade) {
+            throw getAccessDeniedException(ade);
```

Comment:
Can't you just omit this catch block?

Reply:
Yes, of course, technically this would be possible. But it would return a 403 status code and I prefer to hide this information for the user by returning a 404 status.